src/main/java/hudson/plugins/emailext/plugins/content/AbstractEvalContent.java

```diff
@@ -159,15 +160,14 @@ public static boolean isChildOf(final FilePath potentialChild, final FilePath pa
         return parent.act(new IsChildFileCallable(potentialChild));
     }
 
-    private InputStream getManagedFile(String fileName) throws UnsupportedEncodingException {
+    private InputStream getManagedFile(Run<?, ?> run, String fileName) throws UnsupportedEncodingException {
         InputStream stream = null;
         Plugin plugin = Jenkins.get().getPlugin("config-file-provider");
         if (plugin != null) {
             Config config = null;
-            ExtensionList<ConfigProvider> providers = ConfigProvider.all();
-            ConfigProvider provider = providers.get(getProviderClass());
-            assert provider != null;
-            for (Config c : provider.getAllConfigs()) {
+            assert getProviderClass() != null;
```

Comment:
Since this is called below, it would be good to cache this so its not called twice when it doesn't need to be.

Reply:
In fact, I realize that this `assert` is not very relevant, since it was not performed before: it was the `provider` instance for which there was a null-check ; not its class (and not anymore `provider` instance is involved with this new usage).
Are you fine if I remove this useless assertion, then?

Reply:
FYI @slide, I removed this assertion and pushed the change.